runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/KingdomPlugin.java

```diff
@@ -227,13 +228,25 @@ private int estimateCoffer(Instant lastChanged, int lastCoffer)
 
 	private int estimateApproval(Instant lastChanged, int lastApproval)
 	{
-		int daysSince = (int) Duration.between(lastChanged, Instant.now()).toDays();
+		int daysSince = getNumbersOfDaysPassed(lastChanged);
 		float dailyPercentage = isRoyalTroubleCompleted() ? APPROVAL_DECREMENT_ROYAL_TROUBLE : APPROVAL_DECREMENT_BASE;
 
 		lastApproval -= (int) (daysSince * dailyPercentage * MAX_APPROVAL);
 		return Math.max(lastApproval, 0);
 	}
 
+	private int getNumbersOfDaysPassed(Instant lastChanged)
+	{
+		Instant midnight = lastChanged.plus(1, ChronoUnit.DAYS).truncatedTo(ChronoUnit.DAYS);
```

Comment:
I think this reduces to:
```java
	private static int getNumbersOfDaysPassed(Instant lastChanged)
	{
		lastChanged = lastChanged.truncatedTo(ChronoUnit.DAYS);
		var now = Instant.now().truncatedTo(ChronoUnit.DAYS);
		return (int) Duration.between(lastChanged, now).toDays();
	}
```
?

Reply:
Hi! That is correct. I'm not used with pull requests. Do I add your changes in my fork and commit to the pull request? Or can you do something directly from Github to approve you change?

Reply:
You can update the PR. I think your code is adding an extra day (the `+ 1`), do know why that is? Is it required? My code is not.

Reply:
The extra one was required for my code but not yours, I double checked your code and changed the value of lastChanged and the code returned the right number of days that have passed,

I added the extra one because if last changed is 9/16/2023 5PM then midnight variable will be 9/17/2023 Midnight and Instant.Now() could be 9/17/2023 4PM. Then if I understand it correctly then (int) Duration.between(lastChanged, Instant.now()).toDays() would return 0 which is incorrect. As it have not been 24 hours but a day has still passed. I might be incorrect in that assumption but when I tested my code a bit it worked.

Reply:
One thing, should not var now = Instant.now().truncatedTo(ChronoUnit.DAYS) be Instant now instead? There is no need to have it a var?